kit/src/pool/mod.rs

```diff
@@ -13,9 +13,25 @@ pub trait PoolType {
     type Parameters;
     type StrategyContract;
     type SolverContract;
+    type AllocationData;
 
     #[allow(async_fn_in_trait)]
     async fn swap_data(&self, pool_id: eU256, swap_x_in: bool, amount_in: eU256) -> Result<Bytes>;
```

Comment:
Can we change the bool here too?
```rust
pub enum InputToken {
    TokenX,
    TokenY,
}
```

Reply:
This one feels a little awkward i pushed up an approach but not sure if it's too convoluted or not.